modules/experimental-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -133,15 +140,36 @@ export default class MeshLayer extends Layer {
         accessor: 'getPosition',
         update: this.calculateInstancePositions64xyLow
       },
-      instanceRotations: {
-        size: 3,
-        accessor: ['getYaw', 'getPitch', 'getRoll'],
-        update: this.calculateInstanceRotations
-      },
       instanceColors: {
         size: 4,
         accessor: 'getColor',
         defaultValue: [0, 0, 0, 255]
+      },
+      instanceModelMatrix: {
+        size: 16,
+        shaderAttributes: {
+          instanceModelMatrixColumn1: {
+            size: 4,
+            stride: 64,
+            offset: 0
+          },
+          instanceModelMatrixColumn2: {
+            size: 4,
+            stride: 64,
+            offset: 16
+          },
+          instanceModelMatrixColumn3: {
+            size: 4,
+            stride: 64,
+            offset: 32
+          },
+          instanceModelMatrixColumn4: {
+            size: 4,
+            stride: 64,
+            offset: 48
+          }
+        },
+        update: this.calculateInstanceXform
```

Comment:
@Pessimistress @ibgreen What do you think of this API (mostly based Xiaoji's suggestions)? Conceptually, it allows a separation between "layer attributes" (semantically meaningful to the  layer) and the actual "shader attributes" they map to in the vertex shader. And since the layer attribute defines a shared buffer across the shader attributes, it should be usable for any kind of interleaving.

Reply:
Yes this can work.

I don't have a good counterproposal and I don't intend to block, that said I do have some concerns:

* The deck.gl `Attribute` class inherits from the luma.gl `Attribute`, which makes those two classes very coupled. This is already making it quite hard to change things in luma.gl, and now we are adding extra complexity to that Attribute class.
* The `Attribute` class in luma.gl very much represents one attribute. Extending it to multiple attributes here seems like it risks surprising maintainers.
* I wonder if this is always a one-to-many relationship (though that is probably an acceptable limitation).

Finally, this proposal would be stronger if we had a second use case. Perhaps `fp64` attributes could be handled this way, although investigating that is a bit much to ask at this stage.

Reply:
Part of the issue, I think, is naming. The "attributes" in deck shouldn't really be called attributes if we don't want layers managing GL attributes directly. Conceptually, they become something else. Basically a property (or properties) of the layer that will be stored in a single buffer, accessed in the shader by one or more GLSL attributes.

I think Luma attributes should continue to represent one attribute. I'm still getting a handle on the relationship between codebases, but my general feeling is the lowest layer should minimize fanciness. 

Another use case for this kind of structure is attribute interleaving. Is that something we're interested in?